packages/access-api/src/service/index.js

```diff
@@ -33,7 +33,11 @@ export function service(ctx) {
       info: Server.provide(Space.info, async ({ capability, invocation }) => {
         const results = await ctx.models.spaces.get(capability.with)
         if (!results) {
-          return new Failure('Space not found.')
+          return {
+            error: true,
+            status: 404,
+            message: 'Space not found.',
+          }
```

Comment:
we probably should start a error.js file like this https://github.com/web3-storage/ucanto/blob/main/packages/validator/src/error.js and a error.ts for error types in the client

Reply:
this is gonna return a http 200 so i dont think we need the status in there, just a normal `SpaceNotFound extends Failure` with `name` and `message` props.

Reply:
The status is in there as the indication of what kind of error this is. I don't think we should use just a specific 'message' value as the only discriminant of this error case because doing that would make it so that changing the error message string would be a backward-incompatible change from the perspective of any client trying to detect this error case. I'm intentionally trying to add a specific non-`message` [error code](https://en.wikipedia.org/wiki/Error_code) that clients can use to detect this case.

Reply:
exactly, non-http transport should not need to understand http codes. For errors IMO either use the name as the identifier or a code nodejs style `error.code = ERR_SPACE_NOT_FOUND`

```js
// similar to ucanto
if(error.name === 'SpaceNotFound'){}

// similar to nodejs
class SpaceNotFoundError{}
SpaceNotFoundError.code = 'ERR_SPACE_NOT_FOUND'

if(error.code === SpaceNotFoundError.code) {}
```

Reply:
@hugomrdias Thanks, when I first responded I missed that you were encouraging a `name` and thought you were just saying to cut `status` which would make it the same as generic `Failure`. my bad